clang/test/Driver/repo-toolchain-linux-musl.c

```diff
@@ -170,40 +188,40 @@
 // RUN:   -resource-dir=%S/Inputs/resource_dir \
 // RUN:   -ccc-install-dir %S/Inputs/basic_linux_libcxx_tree/bin \
 // RUN:   %s 2>&1 \
-// RUN:   | FileCheck -check-prefix=CHECK006 %s
+// RUN:   | FileCheck -check-prefix=CHECK007 %s
 
-// CHECK006:   "-resource-dir" "[[RESOURCE_DIR:[^"]+]]"
-// CHECK006:  "-L{{/|\\\\}}usr{{/|\\\\}}local{{/|\\\\}}musl{{/|\\\\}}lib"
-// CHECK006:  "-L[[RESOURCE_DIR]]{{/|\\\\}}lib{{/|\\\\}}linux"
-// CHECK006:  "-L{{.*}}basic_linux_libcxx_tree{{/|\\\\}}bin{{/|\\\\}}..{{/|\\\\}}lib"
-// CHECK006:  "-lc++" "-lc++abi"
+// CHECK007:   "-resource-dir" "[[RESOURCE_DIR:[^"]+]]"
+// CHECK007:  "-L{{/|\\\\}}usr{{/|\\\\}}local{{/|\\\\}}musl{{/|\\\\}}lib"
+// CHECK007:  "-L[[RESOURCE_DIR]]{{/|\\\\}}lib{{/|\\\\}}linux"
+// CHECK007:  "-L{{.*}}basic_linux_libcxx_tree{{/|\\\\}}bin{{/|\\\\}}..{{/|\\\\}}lib"
+// CHECK007:  "-lc++" "-lc++abi"
 
 // -----------------------------------------------------------------------------
 // Checking the invoked linker for elf.
 // Passing -musl, -gnu and -linux
 // -----------------------------------------------------------------------------
 // RUN: %clang -### -target x86_64-pc-linux-musl %s 2>&1 \
-// RUN:   | FileCheck -check-prefix=CHECK007 %s
+// RUN:   | FileCheck -check-prefix=CHECK008 %s
 
 // RUN: %clang -### -target x86_64-pc-linux-gnu %s 2>&1 \
-// RUN:   | FileCheck -check-prefix=CHECK007 %s
+// RUN:   | FileCheck -check-prefix=CHECK008 %s
 
 // RUN: %clang -### -target x86_64-pc-linux %s 2>&1 \
-// RUN:   | FileCheck -check-prefix=CHECK007 %s
+// RUN:   | FileCheck -check-prefix=CHECK008 %s
 
-// CHECK007:      "{{.*}}ld"
+// CHECK008:      "{{.*}}ld"
```

Comment:
Won't this test pass for any string ending in “ld” (which means most ELF linkers)‽ Should it end with ”/ld”?

Reply:
On windows, the Driver can’t find the `ld` path, it will use `ld`. For example,
```
"ld" "-pie" "--eh-frame-hdr" "-m" "elf_x86_64" …
```
Therefore, we can’t use `/ld` on Windows.